packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -449,6 +451,10 @@ export function NativeStackView({ state, navigation, descriptors }: Props) {
 
   useInvalidPreventRemoveError(descriptors);
 
+  // if the is a one screen with modal 'presentation'
+  // then all next screens also should be with the same presentation
+  let isForsedModalStack = false;
```

Comment:
I'd change the naming from `isForcedModalStack` to something like `isModal`, since it can either be an automatic modal or specified by the user as a modal, and this is for a screen and not the whole stack.